pkgs/objective_c/lib/src/core.dart

```diff
@@ -123,27 +124,71 @@ class ObjCBlockBase extends _ObjCFinalizable<c.ObjCBlock> {
   void _release(Pointer<c.ObjCBlock> ptr) => c.blockRelease(ptr);
 }
 
-Pointer<c.ObjCBlockDesc> _newBlockDesc() {
+Pointer<c.ObjCBlockDesc> _newBlockDesc(
+    Pointer<NativeFunction<Void Function(Pointer<c.ObjCBlock>)>>
+        disposeHelper) {
   final desc = calloc.allocate<c.ObjCBlockDesc>(sizeOf<c.ObjCBlockDesc>());
   desc.ref.reserved = 0;
   desc.ref.size = sizeOf<c.ObjCBlock>();
   desc.ref.copy_helper = nullptr;
-  desc.ref.dispose_helper = nullptr;
+  desc.ref.dispose_helper = disposeHelper.cast();
   desc.ref.signature = nullptr;
   return desc;
 }
 
-final _blockDesc = _newBlockDesc();
+final _pointerBlockDesc = _newBlockDesc(nullptr);
+final _closureBlockDesc = _newBlockDesc(
+    Native.addressOf<NativeFunction<Void Function(Pointer<c.ObjCBlock>)>>(
+        c.disposeObjCBlockWithClosure));
 
-Pointer<c.ObjCBlock> newBlock(Pointer<Void> invoke, Pointer<Void> target) {
+Pointer<c.ObjCBlock> _newBlock(Pointer<Void> invoke, Pointer<Void> target,
+    Pointer<c.ObjCBlockDesc> descriptor, int disposePort, int flags) {
   final b = calloc.allocate<c.ObjCBlock>(sizeOf<c.ObjCBlock>());
   b.ref.isa = c.NSConcreteGlobalBlock;
-  b.ref.flags = 0;
+  b.ref.flags = flags;
   b.ref.reserved = 0;
   b.ref.invoke = invoke;
   b.ref.target = target;
-  b.ref.descriptor = _blockDesc;
+  b.ref.dispose_port = disposePort;
+  b.ref.descriptor = descriptor;
   final copy = c.blockCopy(b.cast()).cast<c.ObjCBlock>();
   calloc.free(b);
   return copy;
 }
+
+Pointer<c.ObjCBlock> newPointerBlock(
```

Comment:
This function is not intended to be used by users. I just have to expose it so that it can be used by the ffigen bindings. So I've been avoiding writing documentation for any of these functions, to discourage external users from using them (same with `registerName` etc). I'll add a comment that specifically says that.

> Should we have an optional param for disposing something?

If people want that sort of freeing for these sorts of blocks I'll add it, but I haven't heard of a use case for that.